src/Screens/All.tsx

```diff
@@ -1,20 +1,33 @@
 import React from "react";
 
 import { View, StyleSheet, Image, SafeAreaView } from "react-native";
-import PokemonList from "@components/PokemonList";
+import PokemonList, { useScrollHandler } from "@components/PokemonList";
 import usePokemons from "../hooks/usePokemons";
 import { TabScreenProps } from "./types";
+import { useAnimatedStyle, useSharedValue } from "react-native-reanimated";
+import debounce from "lodash/debounce";
+import { SEARCH_HEIGHT } from "../constants";
 
 type Props = TabScreenProps<"All">;
 
 const All = ({ navigation }: Props) => {
+  const [search, _setSearch] = React.useState("");
+  const setSearch = debounce(_setSearch, 500);
+
   const { pokemonIndex, loading, error, fetchMore, loadingMore } =
-    usePokemons();
+    usePokemons(search);
+
+  const searchBar = useSharedValue(0);
+  const scrollHandler = useScrollHandler(searchBar);
```

Comment:
WDYT about changing it to:
```suggestion
  const {searchBar, scrollHandler} = useScrollHandler();
```